airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java

```diff
@@ -114,17 +123,31 @@ private PreparedStatement getPkPreparedStatement(final Connection connection) {
       if (pkLoadStatus == null) {
         LOGGER.info("pkLoadStatus is null");
         final String quotedCursorField = enquoteIdentifier(pkInfo.pkFieldName(), quoteString);
-        final String sql = String.format("SELECT %s FROM %s ORDER BY %s LIMIT %s", wrappedColumnNames, fullTableName,
-            quotedCursorField, chunkSize);
+        final String sql;
+        // We cannot load in chunks for a composite key load, since each field might not have distinct values.
+        if (isCompositeKeyLoad) {
+          sql = String.format("SELECT %s FROM %s ORDER BY %s", wrappedColumnNames, fullTableName,
+              quotedCursorField);
+        } else {
+          sql = String.format("SELECT %s FROM %s ORDER BY %s LIMIT %s", wrappedColumnNames, fullTableName,
+              quotedCursorField, chunkSize);
+        }
         final PreparedStatement preparedStatement = connection.prepareStatement(sql);
         LOGGER.info("Executing query for table {}: {}", tableName, preparedStatement);
         return preparedStatement;
       } else {
         LOGGER.info("pkLoadStatus value is : {}", pkLoadStatus.getPkVal());
         final String quotedCursorField = enquoteIdentifier(pkLoadStatus.getPkName(), quoteString);
-        // Since a pk is unique, we can issue a > query instead of a >=, as there cannot be two records with the same pk.
-        final String sql = String.format("SELECT %s FROM %s WHERE %s > ? ORDER BY %s LIMIT %s", wrappedColumnNames, fullTableName,
-            quotedCursorField, quotedCursorField, chunkSize);
+        final String sql;
+        // We cannot load in chunks for a composite key load, since each field might not have distinct values. Furthermore, we have to issue a >=
+        // query since we may not have processed all of the data associated with the last saved primary key value.
+        if (isCompositeKeyLoad) {
+          sql = String.format("SELECT %s FROM %s WHERE %s >= ? ORDER BY %s", wrappedColumnNames, fullTableName,
```

Comment:
This could lead to duplicate rows for `Incremental | append` but per our conversation, that's something we don't need to optimize for since MySQL CDC customers almost exclusively use Append + deduped.